src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml

```diff
@@ -272,6 +272,77 @@
         </foreach>
     </sql>
 
+    <sql id="applyNumericalDataFilter">
+        <foreach item="dataFilterValue" collection="${data_filter_collection}" open="((" separator=") OR (" close="))">
+            <trim prefix="" prefixOverrides="AND">
+                <!-- Special case: NA -->
+                <if test="dataFilterValue.value eq 'NA'">
+                    AND
+                    <include refid="isAttributeValueNA">
+                        <property name="attribute_value" value="${attribute_value}"/>
+                    </include>
+                </if>
```

Comment:
Maybe we don't need to handle NA here if this SQL is used as the non-NA part? Just curious, it's also fine to have it here. And I think it's also possible to include the NA part as a whole SQL but this is great progress already. Appreciate it.

Reply:
I think this NA part was implemented before we added the additional logic for NA.

You're right. We should probably move the NA filter logic into this same SQL as well. Might be better to deal with it in a separate PR tough.